tests/mocks_gitlab_api.py

```diff
@@ -52,7 +52,18 @@ def register_project(self, numerical_id, full_project_path):
             'projects/' + self.escape_path_in_url(full_project_path),
         response_json={
             'id': numerical_id,
-            'path_with_namespace': full_project_path
+            'path_with_namespace': full_project_path,
+        },
+        helper=True,
+    )
+
+    def register_project_with_mr(self, numerical_id, full_project_path):
```

Comment:
How about keeping just one function and build the JSON from extra arguments.

Something like the following?

```python
def register_project(self, numerical_id, full_project_path, **kwargs):
    response_json_base = {
         'id': numerical_id,
         'path_with_namespace': full_project_path,
    }
    # Not sure about this syntax but Python can merge dictionaries
    response_json = {**response_json_base, **kwargs}
    ...
```

What do you think?

Reply:
Thanks for the suggestion.

It sure does look much cleaner. Now I just have to make it work.

Reply:
This new commit should resolve the problem.

However, in this particular case 'mr_default_target_self' isn't involved
in the kwargs, so I had to add this as an base key in the dictionary.

Reply:
Feel free to drop old code, no need to keep the commented code around if we have Git :-).

> However, in this particular case 'mr_default_target_self' isn't involved
> in the kwargs, so I had to add this as an base key in the dictionary.

Why? What would prevent you to call `register_project(42, '/group/project', mr_default_target_self='self')`?